clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -361,13 +365,21 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
 
             } catch (Exception e) {
                 String desc = channel.socketDescription();
-                if (e instanceof IOException)
+                if (e instanceof ConnectException) {
+                    if (LogRateLimiter.warn(log, "Cannot connect to {}", desc, e,
+                            loggerCount, loggerCountMax))
+                        loggerCount++;
```

Comment:
Also, not sure if we can make this easy, but it'd really be ideal if we could keep the logging line as a single line of code without requiring additional logic by the caller. (Maybe even if this requires allocating a special object to do that.)